### PR TITLE
Test conn on reverse exit before choose entry node

### PR DESCRIPTION
**Pull request by yilunzhang**
Signed-off-by: Yilun <zyl.skysniper@gmail.com>

### CODE REVIEW

File: exit.go [hunk 1]
```diff
@@ -549,6 +549,25 @@ func (te *TunaExit) StartReverse(shouldReconnect bool) error {
...
 		}
 
+		reverseIP := tcpConn.RemoteAddr().(*net.TCPAddr).IP
+		reverseTCP := reverseMetadata.ServiceTcp
+		if len(reverseTCP) > 0 {
+			go func() {
+				net.DialTimeout(tcp, fmt.Sprintf("%s:%d", reverseIP.String(), reverseTCP[0]), defaultReverseTestTimeout)
+			}()
+
+			session.SetDeadline(time.Now().Add(defaultReverseTestTimeout))
```
Comment (bufrr) [inline exit.go:559]: Could we call `SetDeadline` when session init?

File: exit.go [hunk 2]
```diff
@@ -549,6 +549,25 @@ func (te *TunaExit) StartReverse(shouldReconnect bool) error {
...
+
+			_, err = session.AcceptStream()
+			if err != nil {
+				log.Println("Couldn't accept stream test conn from reverse entry:", err)
+				time.Sleep(1 * time.Second)
+				continue
+			}
+
+			session.SetDeadline(time.Time{})
```
Comment (bufrr) [inline exit.go:568]: Is this required to set it back?

File: exit.go (hunk 1 above)
Comment (yilunzhang) [reply exit.go:559]: This is to prevent `session.Accept()` not returning (if entry port is not open)

File: exit.go (hunk 2 above)
Comment (yilunzhang) [reply exit.go:568]: Yes, otherwise `session.Accept()` will always return error after 3s

**Review by bufrr — APPROVED** [review]

**Pull request merged**